### PR TITLE
feat: Rename Stream to EventSource

### PR DESCRIPTION
**Pull request by cynicaljoy**
ENG-6824

### Description
Rename `Stream` to `EventSource`

### Motivation and context
Alignment with the wire protocol and the other drivers

### How was the change tested?
Integration tests

### Change types

* - [ ] Bug fix (non-breaking change that fixes an issue)
* - [ ] New feature (non-breaking change that adds functionality)
* - [x] Breaking change (backwards-incompatible fix or feature)

### Checklist:

* - [ ] My code follows the code style of this project.
* - [ ] My change requires a change to Fauna documentation.
* - [ ] My change requires a change to the README, and I have updated it accordingly.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Fauna/IClient.cs [hunk 1]
```diff
@@ -544,13 +543,13 @@ private async IAsyncEnumerable<Page<T>> PaginateAsyncInternal<T>(
     /// Opens the stream with Fauna and returns an enumerator for the stream events.
     /// </summary>
     /// <typeparam name="T">The type of event data that will be deserialized from the stream.</typeparam>
-    /// <param name="stream">The stream to subscribe to.</param>
+    /// <param name="eventSource">The stream to subscribe to.</param>
```
Comment (pnwpedro) [inline Fauna/IClient.cs:546]:
```suggestion
    /// <param name="eventSource">The event source to subscribe to.</param>
```

**Review by pnwpedro — CHANGES_REQUESTED** [review]
A little nit

File: Fauna.Test/Integration.Tests.cs [hunk 2]
```diff
@@ -524,6 +524,23 @@ public async Task CanResumeStreamWithStreamOptions()
...
 
+    [Test]
+    [Category("Streaming")]
+    public async Task CanOpenStreamWithEventSource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(1)); // prevent runaway test
+        cts.Token.ThrowIfCancellationRequested();
+
+        EventSource eventSource = await _client.GetEventSourceFromQueryAsync(
+            FQL($"StreamingSandbox.all().toStream()"),
```
[the remark this answers is not in the document]
Comment (cynicaljoy) [reply Fauna.Test/Integration.Tests.cs]: thanks, I updated all `toStream` with `eventSource`

**Review by pnwpedro — APPROVED** [review]
lgtm

**Pull request merged**